func.lua

```diff
@@ -259,6 +265,32 @@ function finish()
 
   -- close midi
   STATE.dev:close()
+
+  -- normalize samples if enabled
+  if OPTIONS.post_record_normalize_and_trim then
+    post_record_normalize_and_trim()
+  end
+end
+
+-- normalize and trim the samples
+function post_record_normalize_and_trim()
+  -- ensure background processing is not enabled.
+  -- if background processing is enabled a few bad things could happen:
+  -- normalizing gets skipped or there's a race condition that goes un-noticed
+  -- while the sound data is being treated. 
+  local tmp_background = OPTIONS.background
+  OPTIONS.background = false
+
+  update_status("Normalizing samples...")
```

Comment:
Your comment about the race condition is correct, but my "Process in background" label has led you astray regardless, sorry!

In reality, *both* options process in the background, one just processes "more in the background" (more loop steps before the coroutine yields) than the other. Because of this, in my testing, the race condition you mention still exists and normalizing is skipped when the new option is checked.

This overwriting behavior happens because I lazily made all the background processing dependent on the single `SAMPLE_PROCESSING_PROCESS` global variable in `crunch.lua`.

I think you could solve this by writing a new function that combines the two coroutines so they don't step on each other's toes. 

Something like
```lua
function normalize_and_trim_coroutine()
  -- call processing coroutines serially
  normalize_coroutine()
  trim_coroutine()
end

function normalize_and_trim()
  prep_processing(normalize_and_trim_coroutine)
  SAMPLE_PROCESSING_PROCESS:start()
end
```

Haven't tested this solution! Feel free to solve it another way too.